App/iOS/Delegates/AppDelegate.swift

```diff
@@ -423,10 +424,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
       (SessionRestoreHandler.path, SessionRestoreHandler()),
       (ErrorPageHandler.path, ErrorPageHandler()),
       (ReaderModeHandler.path, ReaderModeHandler(profile: profile)),
-      (SNSDomainHandler.path, SNSDomainHandler()),
       (IPFSSchemeHandler.path, IPFSSchemeHandler())
-    ]
-    
+    ] + Web3Service.allCases.map {
+      (Web3DomainHandler.path(for: $0), Web3DomainHandler(for: $0))
```

Comment:
:o How many custom handlers are we registering here?

Reply:
3 cases in this PR (Ethereum, Ethereum Offchain, Solana), one more added in https://github.com/brave/brave-ios/pull/7165 for Unstoppable Domains

Reply:
Storing service in query so only one `Web3DomainHandler` added now: [9cbb5a6](https://github.com/brave/brave-ios/pull/7153/commits/9cbb5a6b85cdeeca6b425f2fd0ba3329d9d09579)